hy/core/macros.hy

```diff
@@ -20,6 +20,20 @@
          ~@(interleave (repeat name) rest))
      ~name))
 
+
+(defmacro assoc [coll k1 v1 &rest other-kvs]
+  (if (odd? (len other-kvs))
+    (macro-error (last other-kvs) "`assoc` takes an odd number of arguments"))
+  (setv c (if other-kvs (gensym "c") coll))
+  `(setv
+    ~@(if other-kvs [c coll] [])
```

Comment:
Since #1349, splice works on any False value. This could be `~@(if other-kvs [c coll])` (or `when`).